File: 1-TwoSum/1-TwoSum.java
// Last updated: 22/07/2026, 23:31:20
1class Solution {
2    public int[] twoSum(int[] nums, int target) {
3        HashMap<Integer, Integer> map = new HashMap<>();
4        for (int i = 0; i < nums.length; i++) {
5            int diff = target - nums[i];
6            if (map.containsKey(diff)) {
7                return new int[] { map.get(diff), i };
8            }
9            map.put(nums[i], i);
10        }
11        return new int[0];
12    }
13}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 17/07/2026, 00:11:18
1class Solution {
2public:
3    vector<int> twoSum(vector<int>& nums, int target) {
4
5        std::unordered_map<int,int> ValueToIndexMap;
6
7        for(int currentIndex=0;currentIndex < nums.size();currentIndex++)
8        {
9            int currentValue = nums[currentIndex];
10            int requiredValue = target - currentValue;
11
12            if(ValueToIndexMap.find(requiredValue) != ValueToIndexMap.end())
13            {
14                return {ValueToIndexMap[requiredValue],currentIndex};
15            }
16            ValueToIndexMap[currentValue] = currentIndex;
17        }
18        return {-1,-1};
19        
20    }
21};